src/realm-settings/key-providers/aes-generated/AESGeneratedModal.tsx

```diff
@@ -0,0 +1,65 @@
+import React from "react";
+import { AlertVariant, Modal, ModalVariant } from "@patternfly/react-core";
+import { useTranslation } from "react-i18next";
+
+import type ComponentRepresentation from "keycloak-admin/lib/defs/componentRepresentation";
+import { AESGeneratedForm } from "./AESGeneratedForm";
+import { useAlerts } from "../../../components/alert/Alerts";
+import { useAdminClient } from "../../../context/auth/AdminClient";
+import { useRealm } from "../../../context/realm-context/RealmContext";
+
+type AESGeneratedModalProps = {
+  providerType?: string;
+  handleModalToggle?: () => void;
+  refresh?: () => void;
```

Comment:
All of these types are always passed and can thus be made required.
```suggestion
  providerType: string;
  handleModalToggle: () => void;
  refresh: () => void;
```